x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/create_trusted_app_form.tsx

```diff
@@ -312,6 +331,33 @@ export const CreateTrustedAppForm = memo<CreateTrustedAppFormProps>(
       });
     }, []);
 
+    const handlePolicySelectChange: EffectedPolicySelectProps['onChange'] = useCallback(
+      (selection) => {
+        setSelectedPolicies(() => selection);
+
+        let newEffectedScope: EffectScope;
+
+        if (selection.isGlobal) {
+          newEffectedScope = {
+            type: 'global',
+          };
+        } else {
+          newEffectedScope = {
+            type: 'policy',
+            policies: selection.selected.map((policy) => policy.id),
+          };
+        }
```

Comment:
I'm always going back and forth between making component interface domain agnostic and communicating in domain entities. I wonder if the decoupling pays off here. Will we be able to reuse the EffectedPolicySelect in other contexts?

Reply:
I would imagine that the ability to select a list of Policies or "apply to all" may be needed in the future with other features (especialloy with artifacts type of features), so I made that component "pure" and have the data be fed in from there.

Reply:
Clear